homeassistant/components/roborock/config_flow.py

```diff
@@ -0,0 +1,147 @@
+"""Config flow for Roborock."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from roborock.api import RoborockClient
+from roborock.containers import UserData
+from roborock.exceptions import RoborockException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_BASE_URL, CONF_ENTRY_CODE, CONF_USER_DATA, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class RoborockFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Roborock."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._username = None
+        self._errors: dict[str, str] = {}
+        self._client: RoborockClient | None = None
+        self._auth_method: str | None = None
+
+    async def async_step_user(
```

Comment:
These forms show one step and switch between another step, which is not allowed. While rendering step user it sends the user to step email, and instead it needs to stay on step user. These steps should be combined.  To advance to the next step, call the next step and await on it.